### PR TITLE
backport: Merge bitcoin#21712,21192, 21695,21244,21595

### PR DESCRIPTION
**Pull request by vijaydasmp**
backport: Merge bitcoin#21712,21192, 21695,21244,21595

### CODE REVIEW

File: src/qt/test/rpcnestedtests.cpp [hunk 1]
```diff
@@ -13,8 +13,10 @@
 
 #include <llmq/context.h>
 
-#include <QDir>
-#include <QtGlobal>
+#include <QTest>
+
+#include <string>
+#include <stdexcept>
```
Comment (UdjinM6) [inline src/qt/test/rpcnestedtests.cpp:19]: 21557: missing changes in `rpcNestedTest_rpc` probably due to missing backports

File: src/net_processing.cpp [hunk 2]
```diff
@@ -3158,6 +3162,7 @@ void PeerManagerImpl::ProcessMessage(
         if (pfrom.fSuccessfullyConnected) {
             // Disconnect peers that send SENDADDRV2 message after VERACK; this
             // must be negotiated between VERSION and VERACK.
+            LogPrint(BCLog::NET, "sendaddrv2 received after verack from peer=%d; disconnecting\n", pfrom.GetId());
```
Comment (UdjinM6) [inline src/net_processing.cpp:3165]: 20724: missing changes after `if (msg_type == NetMsgType::VERACK)` probably due to missing backports

Comment (vijaydasmp) [reply src/net_processing.cpp:3165]: Got it, missing 20146 is the missing one

File: src/qt/test/rpcnestedtests.cpp (hunk 1 above)
Comment (vijaydasmp) [reply src/qt/test/rpcnestedtests.cpp:19]: one of the dependent backport is in https://github.com/dashpay/dash/pull/5936,  , will merge after this

File: src/qt/test/rpcnestedtests.cpp [hunk 3]
```diff
@@ -28,7 +30,7 @@ static RPCHelpMan rpcNestedTest_rpc()
         },
         {},
         RPCExamples{""},
-        [](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue {
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue {
             return request.params.write(0, 0);
```
Comment (vijaydasmp) [inline src/qt/test/rpcnestedtests.cpp:34]: add this change after importing missing backports

File: src/net_processing.cpp [hunk 4]
```diff
@@ -3108,7 +3112,10 @@ void PeerManagerImpl::ProcessMessage(
     bool fBlocksOnly = pfrom.IsBlockRelayOnly();
 
     if (msg_type == NetMsgType::VERACK) {
-        if (pfrom.fSuccessfullyConnected) return;
+        if (pfrom.fSuccessfullyConnected) {
```
Comment (vijaydasmp) [inline src/net_processing.cpp:3115]: 20724: added this missing change after including backport bitcoin#20146

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by vijaydasmp** [comment]
Hello @knst @PastaPastaPasta , requesting review

File: src/bitcoin-cli.cpp [hunk 5]
```diff
@@ -62,6 +63,7 @@ static void SetupCliArgs(ArgsManager& argsman)
     argsman.AddArg("-conf=<file>", strprintf("Specify configuration file. Relative paths will be prefixed by datadir location. (default: %s)", BITCOIN_CONF_FILENAME), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
     argsman.AddArg("-datadir=<dir>", "Specify data directory", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
     argsman.AddArg("-generate", strprintf("Generate blocks immediately, equivalent to RPC getnewaddress followed by RPC generatetoaddress. Optional positional integer arguments are number of blocks to generate (default: %s) and maximum iterations to try (default: %s), equivalent to RPC generatetoaddress nblocks and maxtries arguments. Example: dash-cli -generate 4 1000", DEFAULT_NBLOCKS, DEFAULT_MAX_TRIES), ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
+    argsman.AddArg("-addrinfo", "Get the number of addresses known to the node, per network and total.", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
```
Comment (knst) [inline src/bitcoin-cli.cpp:66]: nit: Consider to do bitcoin/bitcoin#21753 also which adds docs for this option.

Can be done by next PR, not a blocker to get merged.

**Review by knst — APPROVED** [review]
utACK

**Pull request merged**